### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by bkostanetskyi**

### CODE REVIEW

File: src/main/java/mate/academy/books/BooksApplication.java [hunk 1]
```diff
@@ -10,4 +13,11 @@ public static void main(String[] args) {
         SpringApplication.run(BooksApplication.class, args);
     }
 
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return args -> {
+            Book book1 = new Book();
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/BooksApplication.java]: don't use numbers for naming

File: src/main/java/mate/academy/books/BooksApplication.java [hunk 2]
```diff
@@ -10,4 +13,11 @@ public static void main(String[] args) {
...
+            book1.setTitle("Book 1");
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/BooksApplication.java]: you should create book and call BookService methods

File: src/main/java/mate/academy/books/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
+package mate.academy.books.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+
+@Entity
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/model/Book.java:13]:
```suggestion
@Entity
@Table....
@Getter
@Setter
```

File: src/main/java/mate/academy/books/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    @Column(nullable = false)
+    private BigDecimal price;
+
+    @Column(length = 1000)
+    private String description;
+
+    @Column
+    private String coverImage;
+
+    // Getter-и та Setter-и
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/model/Book.java]: remove getters and setters

File: src/main/java/mate/academy/books/model/Book.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    @Column(nullable = false, unique = true)
+    private String isbn;
+
+    @Column(nullable = false)
+    private BigDecimal price;
+
+    @Column(length = 1000)
+    private String description;
+
+    @Column
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/model/Book.java]:
```suggestion
```

File: src/main/java/mate/academy/books/repository/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import mate.academy.books.model.Book;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final List<Book> books = new ArrayList<>();
+
+    @Override
+    public Book save(Book book) {
+        books.add(book);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/repository/BookRepositoryImpl.java]: implement adding book to database

File: src/main/java/mate/academy/books/repository/BookRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return new ArrayList<>(books);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/repository/BookRepositoryImpl.java]: implement find all books from database

File: src/main/java/mate/academy/books/service/BookServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
+package mate.academy.books.service;
+
+import java.util.List;
+import mate.academy.books.model.Book;
+import mate.academy.books.repository.BookRepository;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/service/BookServiceImpl.java:13]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/mate/academy/books/service/BookServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/service/BookServiceImpl.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/books/repository/BookRepositoryImpl.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+            if (session != null) {
+                session.close();
+            }
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return sessionFactory.getCurrentSession()
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/repository/BookRepositoryImpl.java]: add try catch

File: src/main/java/mate/academy/books/BooksApplication.java [hunk 11]
```diff
@@ -1,13 +1,29 @@
...
         SpringApplication.run(BooksApplication.class, args);
     }
 
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return args -> {
+            Book book = new Book();
+            book.setTitle("Book 1");
+
+            bookRepository.save(book);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/BooksApplication.java]: you should create Book, save it and call findAll books here

File: src/main/java/mate/academy/books/repository/BookRepositoryImpl.java [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(book);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/repository/BookRepositoryImpl.java:31]: throw new DataProcessingException

File: src/main/java/mate/academy/books/repository/BookRepositoryImpl.java [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
+import java.util.List;
+import mate.academy.books.model.Book;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/repository/BookRepositoryImpl.java:13]:
```suggestion
@Repository
@RequiredArgsConstructor
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/books/BooksApplication.java (hunk 11 above)
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/BooksApplication.java]: Comment wasn't fixed.

File: src/main/java/mate/academy/books/repository/BookRepositoryImpl.java (hunk 12 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/books/repository/BookRepositoryImpl.java:31]: comment wasn't fixed

File: src/main/java/mate/academy/books/repository/BookRepositoryImpl.java [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+                session.close();
+            }
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        try {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/repository/BookRepositoryImpl.java]: use try with resources

File: src/main/java/mate/academy/books/repository/BookRepositoryImpl.java [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+            if (session != null) {
+                session.close();
+            }
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/repository/BookRepositoryImpl.java]: remove

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments
If you have some questions please ask in chat

File: src/main/java/mate/academy/books/BooksApplication.java [hunk 16]
```diff
@@ -1,13 +1,32 @@
...
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/BooksApplication.java]: You should call BookService methods, I mentioned this here https://github.com/bkostanetskyi/Books/pull/1#discussion_r1763563436

Implement this comment and run you application locally to get the result

File: src/main/java/mate/academy/books/repository/BookRepositoryImpl.java [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/repository/BookRepositoryImpl.java:31]: throw new DataProcessingException

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/books/BooksApplication.java [hunk 18]
```diff
@@ -1,13 +1,38 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
 
 @SpringBootApplication
 public class BooksApplication {
+    @Autowired
+    private BookRepository bookRepository;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/BooksApplication.java]:
```suggestion
    private BookService bookService;
```

File: src/main/java/mate/academy/books/BooksApplication.java [hunk 19]
```diff
@@ -1,13 +1,38 @@
...
+        return args -> {
+            Book book = new Book();
+            book.setTitle("Effective Java");
+            book.setAuthor("Joshua Bloch");
+            book.setIsbn("978-0134685991");
+            book.setPrice(new BigDecimal("45.00"));
+            book.setDescription("A comprehensive guide to best practices in Java programming.");
+            book.setCoverImage("effective_java_cover.jpg");
+
+            bookRepository.save(book);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/BooksApplication.java]:
```suggestion
            bookService.save(book);
```

File: src/main/java/mate/academy/books/BooksApplication.java [hunk 20]
```diff
@@ -1,13 +1,38 @@
...
+
+            List<Book> books = bookRepository.findAll();
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/BooksApplication.java]:
```suggestion
            List<Book> books = bookService.findAll();
```

see the result on console

File: src/main/java/mate/academy/books/repository/BookRepositoryImpl.java [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+            throw new DataProcessingException("Error occurred during data processing", e);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/repository/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can't save the book", e);
```

File: src/main/java/mate/academy/books/repository/BookRepositoryImpl.java [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+                session.close();
+            }
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from Book", Book.class).list();
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/repository/BookRepositoryImpl.java]:
```suggestion
            return session.createQuery("from Book", Book.class).getResultList();
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Review by sarakhmen — APPROVED** [review]
lgtm!

**Pull request merged**